crates/ruff/src/rules/flake8_return/rules.rs

```diff
@@ -354,10 +369,75 @@ fn unnecessary_assign(checker: &mut Checker, stack: &Stack, expr: &Expr) {
             return;
         }
 
-        checker.diagnostics.push(Diagnostic::new(
-            UnnecessaryAssign,
-            Range::from_located(expr),
-        ));
+        let mut diagnostic = Diagnostic::new(UnnecessaryAssign, Range::from_located(expr));
+
+        if checker.patch(diagnostic.kind.rule()) {
+            if let Some(assign_expr) = stack.assign_values.get(id.as_str()) {
+                if let Some(assign_locations) = stack.assigns.get(id.as_str()) {
+                    let expr_loc = expr.end_location.unwrap();
+                    if let Some(last_assign_loc) = assign_locations
```

Comment:
When would this be `None`?

Reply:
If you return an undefined variable (a problem in itself, but shouldn't cause a panic). The second `if let` can be removed however the third one is also important because it would fail if the variable was assigned to, but only *after* the return statement (similar problem). In my earlier testing it appeared that every assignment would be in the vector even if they were after the return statement (which was against an assumption I had made and was causing that infinite loop error because it was slicing the wrong way)